aiida_testing/export_cache/_fixtures.py

```diff
@@ -0,0 +1,375 @@
+        process_class=None,
+        label: str = '',
+        data_dir: ty.Union[str, pathlib.Path] = 'data_dir',
+    ):
+        """
+        Function, which checks if a aiida export for a given Process builder exists,
+        if it does it imports the aiida graph and runs the builder with caching.
+        If the cache does not exists, it still runs the builder but creates an
+        export afterwards.
+
+        Inputs:
+
+        builder : AiiDA Process builder class,
+        data_dir: optional
+            Absolute path of the directory where the exported workchain graphs are
+            stored.
+
+        ignore_nodes : list string, ignore input nodes with these labels/link labels to ignore in hash.
+        # needed?
+        """
+
+        cache_exists = False
+        if data_dir == 'data_dir':
+            cwd = pathlib.Path(os.getcwd())  # Might be not the best idea.
+            data_dir = (cwd / 'data_dir')  # TODO: get from config?
+
+        bui_hash, input_nodes = get_hash_process(builder)
+
+        if process_class is None:  # and isinstance(builder, dict):
+            process_class = builder.process_class  # type: ignore
+            # we assume ProcessBuilder, since type(ProcessBuilder) is abc
+        #else:
+        #    raise TypeError(
+        #        'builder has to be of type ProcessBuilder if no process_class is specified'
+        #    )
+        name = label + str(process_class).split('.')[-1].strip("'>") + '-nodes-' + bui_hash
+        print(name)
+
+        # check existence
+        full_import_path = str(data_dir) + '/' + name + '.tar.gz'
+        print(full_import_path)
+        cache_path = pathlib.Path(full_import_path)
+        if cache_path.exists():
+            cache_exists = True
+
+        if cache_exists:
+            # import data from previous run to use caching
+            load_cache(path_to_cache=full_import_path)
+
+        # now run process/workchain whatever
+        with enable_caching():  # should enable caching globally in this python interpreter
+            #yield #test is running
+            #res, resnode = run_get_node(builder)
+            res, resnode = run_get_node(process_class, **builder)
+
+        # This is executed after the test
+        if not cache_exists:
+            # TODO create datadir if not existent
+
+            # in case of yield:
+            # is the db already cleaned?
+            # since we do not the stored process node we try to get it from the inputs,
+            # i.e to which node they are all connected, with the lowest common pk
+            union_pk: ty.Set[int] = set()
+            for node in input_nodes:
+                pks = {ent.node.pk for ent in node.get_outgoing().all()}
+                union_pk = union_pk.union(pks)
+            if len(union_pk) != 0:
+                process_node_pk = min(union_pk)
+                #export data to reuse it later
+                export_cache(node=load_node(process_node_pk), savepath=full_import_path)
```

Comment:
This logic seems problematic. Suppose we prepare the inputs to a workflow with a `calcfunction` which then has a lower pk. Then the export would be done for the calcfunction and not the workflow.

Why don't we just use `export_cache(node=resnode, ...)`? Then we are sure to export the workflow.

Reply:
This was implement because I tried no to use run_get_node within the fixture. if we say it is ok that the workchain is run within the fixture (fixture should only prepare and clean up), this part becomes obsolete. This fixture is therefore like a monkeypatch for run_get_node

Reply:
OK but this did not work for [this test](https://github.com/JuDFTteam/aiida-kkr/blob/develop/aiida_kkr/tests/workflows/test_kkrimp_sub_wc.py) where I prepare an input to the workflow with a `calcfunction` which therefore has a lower pk. Then only the `calcfunction`  is exported but not the workflow nodes itself.

But maybe this can be fixed with an option for the export.